packages/flutter/lib/src/rendering/shifted_box.dart

```diff
@@ -257,6 +257,9 @@ class OneChildLayoutDelegate {
 
   /// Returns the position where the child should be placed given the size of this object and the size of the child.
   Point getPositionForChild(Size size, Size childSize) => Point.origin;
+
+  /// Override this method to return true when the child needs to be laid out.
+  bool shouldRelayout() => false;
```

Comment:
This function should be passed the oldDelegate so that it can make a meaningful decision.